packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/editor/CachedQueryService.kt

```diff
@@ -83,18 +86,22 @@ class CachedQueryService(
                     for (entry in cachedQueriesByNamespace.entries) {
                         val queries = entry.value
                         entry.setValue(
-                            queries.filter { it.get() != null }.toSet()
+                            queries.filter { it.get() != null }
+                                .filter {
+                                    runCatching { it.get()?.source?.containingFile }.isSuccess
+                                }
+                                .distinctBy { it.get()?.source?.textOffset }.toSet()
                         )
                     }
 
                     // now add the current query to the cache
-                    cachedQueriesByNamespace.compute(collRef.namespace) { ns, queries ->
+                    cachedQueriesByNamespace.compute(namespaceOfQuery) { ns, queries ->
                         (queries ?: emptySet()) + WeakReference(parsedAst)
                     }
                 }
             }
 
-            CachedValueProvider.Result.create(parsedAst, attachment)
+            CachedValueProvider.Result.create(parsedAst, attachment, toolbar)
```

Comment:
I've added this because when the toolbar changes, we need to refresh the cache. This is relevant for Spring Criteria. I'm aware that we are getting rid of the toolbar, but we will still need something like this for the sidepanel.